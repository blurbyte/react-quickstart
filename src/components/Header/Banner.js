import styled from 'styled-components';

const Banner = styled.header`
  position: fixed;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 6.4rem;
  top: 0;
  left: 0;
  width: 100%;
  border-bottom: 1px solid #908e89;
  height: 6.4rem;
  box-shadow: 0 4px 4px -1px rgba(40,40,40,0.1);
  background: #fff;
`;

export default Banner;
