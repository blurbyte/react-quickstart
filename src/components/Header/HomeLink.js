import { Link } from 'react-router-dom';
import styled from 'styled-components';

const HomeLink = styled(Link)`
  height: 4rem;
  border-right: 1px solid #908e89;
  display: inline-block;
  padding-top: 0.75rem;
  padding-right: 1.4rem;
  margin-right: 1.4rem;
`;

export default HomeLink;
