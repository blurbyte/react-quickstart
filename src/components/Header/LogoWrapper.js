import styled from 'styled-components';

const LogoWrapper = styled.header`
  display: inline-flex;
  align-items: center;
`;

export default LogoWrapper;
