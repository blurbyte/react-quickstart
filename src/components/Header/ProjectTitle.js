import styled from 'styled-components';

const ProjectTitle = styled.p`
  font-family: 'OpenSans Light', Arial, Helvetica, sans-serif;
  color: #908e89;
  font-size: 1.4rem;
  text-transform: uppercase;
  cursor: default;
`;

export default ProjectTitle;
