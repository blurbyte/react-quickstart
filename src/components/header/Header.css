.banner {
  position: fixed;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 6.4rem;
  top: 0;
  left: 0;
  width: 100%;
  border-bottom: 1px solid #908e89;
  height: 6.4rem;
  box-shadow: 0 4px 4px -1px rgba(40,40,40,0.1);
  background: #fff;
}

.logo {
  display: inline-flex;
  align-items: center;
}

.link {
  height: 4rem;
  border-right: 1px solid #908e89;
  display: inline-block;
  padding-top: 0.75rem;
  padding-right: 1.4rem;
  margin-right: 1.4rem;
}

.description {
  font-family: 'OpenSans Light', Arial, Helvetica, sans-serif;
  color: #908e89;
  font-size: 1.4rem;
  text-transform: uppercase;
  cursor: default;
}
