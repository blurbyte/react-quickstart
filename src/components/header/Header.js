import React from 'react';
import {IndexLink} from 'react-router';

//components
import HeaderCounterContainer from './HeaderCounterContainer';

//import logo
import {BlurbyteLogo} from '../common/Icons';

//css modules styles, using CommonJS syntax to bypass eslint errors
const styles = require('./Header.css');

const Header = () => (
  <header className={styles.banner} role="banner">
    <div className={styles.logo}>
      <IndexLink className={styles.link} to="/"><BlurbyteLogo /></IndexLink>
      <p className={styles.description}>React Quickstart</p>
    </div>
    <HeaderCounterContainer />
  </header>
);

export default Header;
