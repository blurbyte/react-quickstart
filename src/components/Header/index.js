import React from 'react';

// components
import Banner from './Banner';
import LogoWrapper from './LogoWrapper';
import HomeLink from './HomeLink';
import ProjectTitle from './ProjectTitle';
import Logo from 'components/Icons/Logo';

// container
import HeaderCounter from 'containers/HeaderCounter';

const Header = () =>
  <Banner>
    <LogoWrapper>
      <HomeLink to="/">
        <Logo />
      </HomeLink>
      <ProjectTitle>React Quickstart</ProjectTitle>
    </LogoWrapper>
    <HeaderCounter />
  </Banner>;

export default Header;
