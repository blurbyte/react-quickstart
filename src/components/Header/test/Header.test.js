import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../index';

//mocked components
jest.mock('../Banner', () => 'Banner');
jest.mock('../LogoWrapper', () => 'LogoWrapper');
jest.mock('../HomeLink', () => 'HomeLink');
jest.mock('../ProjectTitle', () => 'ProjectTitle');
jest.mock('components/Icons/Logo', () => 'Logo');
jest.mock('containers/HeaderCounter', () => 'HeaderCounter');

describe('<Header />', () => {
  it('should render correctly', () => {
    // component init
    const component = renderer.create(
      <Header />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
