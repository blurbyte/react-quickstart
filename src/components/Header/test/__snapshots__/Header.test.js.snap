// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `
<Banner>
  <LogoWrapper>
    <HomeLink
      to="/"
    >
      <Logo />
    </HomeLink>
    <ProjectTitle>
      React Quickstart
    </ProjectTitle>
  </LogoWrapper>
  <HeaderCounter />
</Banner>
`;
